### PR TITLE
Improve send robots to safe zone during localization mission

### LINKED ISSUE #1461: Clicking the send robots to safe zone button should always freeze the mission queue
**Describe the bug**
When clicking the send robots to safe zone button during a lozalization mission, the robot will not be localized and current area is null. As a consequence, there is no safe position for the robot to go to and the robot continues the localization and runs the missions in the queue normally.

Even if the robot is not localized, when clicking the send robots to safe zone, the mission queue should be frozen and no missions should run.

**To Reproduce**
Press the send robots to safe zone button before the localization mission updated the current area of the robot.

**Expected behavior**
When clicking the send robots to safe zone, the mission queue should be frozen and no missions should run regardless if the robot being localized or not.

**Screenshots**

### PR DESCRIPTION
**Pull request by mrica-equinor**
Closes #1461

### CODE REVIEW

**Review by andchiind — APPROVED** [review]
Haven't managed to make it fail, LGTM

File: backend/api/EventHandlers/MissionEventHandler.cs [hunk 1]
```diff
@@ -80,6 +81,11 @@ private async void OnMissionRunCreated(object? sender, MissionRunCreatedEventArg
                     _scheduleLocalizationSemaphore.Release();
                     return;
                 }
+                if (await MissionService.OngoingLocalizationMissionRunExists(missionRun.Robot.Id))
```
Comment (oysand) [inline backend/api/EventHandlers/MissionEventHandler.cs:84]: Can this be combined with pending above?

File: backend/api/EventHandlers/MissionEventHandler.cs [hunk 2]
```diff
@@ -206,22 +208,70 @@ private async void OnEmergencyButtonDepressedForRobot(object? sender, EmergencyB
...
+                return null;
+            }
+
+            var area = await AreaService.ReadById(robot.CurrentArea!.Id);
+            if (area == null)
+            {
+                _logger.LogError("Could not find area with ID {AreaId}", robot.CurrentArea!.Id);
+                SignalRService.ReportSafeZoneFailureToSignalR(robot, $"Robot {robot.Name} was not correctly localised. Could not find area {robot.CurrentArea.Name}");
+            }
+            return area ?? null;
```
Comment (oysand) [inline backend/api/EventHandlers/MissionEventHandler.cs:273]:
```suggestion
            var area = await AreaService.ReadById(robot.CurrentArea!.Id);
            if (area == null)
            {
                _logger.LogError("Could not find area with ID {AreaId}", robot.CurrentArea!.Id);
                SignalRService.ReportSafeZoneFailureToSignalR(robot, $"Robot {robot.Name} was not correctly localised. Could not find area {robot.CurrentArea.Name}");
                return null;
            }
            return area;
```

**Review by oysand — APPROVED** [review]
LGTM

**Review by oysand — CHANGES_REQUESTED** [review]
Woops, added for wrong PR. Still need the tests discussed

File: backend/api/EventHandlers/MissionEventHandler.cs [hunk 3]
```diff
@@ -149,26 +155,20 @@ private async void OnEmergencyButtonPressedForRobot(object? sender, EmergencyBut
...
-            {
-                _logger.LogError("Could not find area with ID {AreaId}", robot.CurrentArea!.Id);
-                SignalRService.ReportSafeZoneFailureToSignalR(robot, $"Robot {robot.Name} was not correctly localised. Could not find area {robot.CurrentArea.Name}");
-                return;
-            }
-
             try { await MissionScheduling.FreezeMissionRunQueueForRobot(e.RobotId); }
             catch (RobotNotFoundException) { return; }
 
+            var area = await FindRobotArea(robot.Id);
```
Comment (oysand) [inline backend/api/EventHandlers/MissionEventHandler.cs:161]: Currently it is abit unclear from the name that this function also checks what area the robot will be localized in, could the name be more explicit? Something along the lines of FindCurrentRobotAreaOrTheOneItWillBeLocalizedIn, just better formulated?

Comment (oysand) [reply backend/api/EventHandlers/MissionEventHandler.cs:161]: Maybe FindRelevantRobotAreaForSafePositionMission?

File: backend/api/EventHandlers/MissionEventHandler.cs [hunk 4]
```diff
@@ -206,22 +206,71 @@ private async void OnEmergencyButtonDepressedForRobot(object? sender, EmergencyB
...
+            if (robot == null)
+            {
+                _logger.LogError("Robot with ID: {RobotId} was not found in the database", robotId);
+                return null;
+            }
+
+            if (!await LocalizationService.RobotIsLocalized(robotId))
+            {
+
+                if (await MissionService.OngoingLocalizationMissionRunExists(robotId))
```
Comment (oysand) [inline backend/api/EventHandlers/MissionEventHandler.cs:246]: Should we also check for pending localization mission?

**Review by tsundvoll — APPROVED** [review]

**Review by oysand — APPROVED** [review]
LGTM

**Pull request merged**